core/field_variable.js

```diff
@@ -225,20 +219,15 @@ Blockly.FieldVariable.prototype.saveState = function() {
  * @package
  */
 Blockly.FieldVariable.prototype.loadState = function(state) {
-  if (Blockly.FieldVariable.prototype.loadState === this.loadState &&
-      Blockly.FieldVariable.prototype.fromXml !== this.fromXml) {
-    this.fromXml(Blockly.Xml.textToDom(/** @type {string} */ (state)));
-    return;
+  if (!this.loadLegacyState(Blockly.Field, state)) {
+    // This is necessary so that blocks in the flyout can have custom var names.
+    var variable = Blockly.Variables.getOrCreateVariablePackage(
+        this.sourceBlock_.workspace,
+        state['id'] || null,
+        state['name'],
+        state['type'] || '');
+    this.setValue(variable.getId());
   }
```

Comment:
Something I learned from the Go styleists which I think is a generally applicable principle is that it's usually easier to read a function or loop that [gets rid of the special cases at the top and leaves the main body unindented](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow):
```JS
function foo() {
  if (specialCondition)  return;
  doUsualThing();
}
```
This convention is not so universally observed in JavaScript, but I think it can often be helpful nonetheless.